projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -0,0 +1,249 @@
+<?php
+/**
+ * Base Admin Menu file.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+use Automattic\Jetpack\Status;
+
+/**
+ * Class Base_Admin_Menu
+ */
+abstract class Base_Admin_Menu {
+	/**
+	 * Holds class instances.
+	 *
+	 * @var array
+	 */
+	protected static $instances;
+
+	/**
+	 * Whether the current request is a REST API request.
+	 *
+	 * @var bool
+	 */
+	protected $is_api_request = false;
+
+	/**
+	 * Domain of the current site.
+	 *
+	 * @var string
+	 */
+	protected $domain;
+
+	/**
+	 * Base_Admin_Menu constructor.
+	 */
+	protected function __construct() {
+		add_action( 'admin_menu', array( $this, 'set_is_api_request' ), 99999 );
+		add_action( 'admin_menu', array( $this, 'reregister_menu_items' ), 99998 );
```

Comment:
In order to maintain the previous behavior, `set_is_api_request` should run before `reregister_menu_items`. 
```suggestion
		add_action( 'admin_menu', array( $this, 'set_is_api_request' ), 99998 );
		add_action( 'admin_menu', array( $this, 'reregister_menu_items' ), 999989);
```

Reply:
This will also fix the current test failures:

```
 1) Test_Admin_Menu::test_get_instance
  Failed asserting that 99998 is identical to 99999.
  
  /tmp/wordpress-latest/src/wp-content/plugins/jetpack/tests/php/modules/masterbar/test-class-admin-menu.php:96
  
  2) Test_Atomic_Admin_Menu::test_get_instance
  Failed asserting that 99998 is identical to 99999.
  
  /tmp/wordpress-latest/src/wp-content/plugins/jetpack/tests/php/modules/masterbar/test-class-atomic-admin-menu.php:103
  
  3) Test_WPcom_Admin_Menu::test_get_instance
  Failed asserting that 99998 is identical to 99999.
  
  /tmp/wordpress-latest/src/wp-content/plugins/jetpack/tests/php/modules/masterbar/test-class-wpcom-admin-menu.php:103
```